catalyst_voices/lib/widgets/menu/voices_menu.dart

```diff
@@ -0,0 +1,172 @@
+import 'package:catalyst_voices_assets/catalyst_voices_assets.dart';
+import 'package:flutter/material.dart';
+
+/// A menu of the app that
+/// can be also us as a cascade.
+class VoicesMenu extends StatelessWidget {
+  /// Menu items passed as models, can be nested.
+  final List<MenuItem> menuItems;
+
+  /// The widget that is clicked to open menu.
+  final Widget child;
+
+  /// The callback called when the menu item is tapped.
+  final ValueChanged<String>? onTap;
+
+  /// The default constructor for the [VoicesMenu].
+  const VoicesMenu({
+    super.key,
+    required this.menuItems,
+    required this.child,
+    this.onTap,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: MenuBar(
+        style: const MenuStyle(
+          elevation: WidgetStatePropertyAll<double>(0),
+        ),
+        children: [
+          SubmenuButton(
+            menuChildren: [...menuItems.map(_mapItemToButton)],
+            menuStyle: const MenuStyle(alignment: Alignment.centerRight),
+            style: MenuItemButton.styleFrom(shadowColor: Colors.transparent),
+            child: child,
+          ),
+        ],
+      ),
+    );
+  }
+
+  _MenuButton _mapItemToButton(MenuItem item) {
+    return _MenuButton(
+      label: item.label,
+      icon: item.icon,
+      showDivider: item.showDivider,
+      enabled: item.enabled,
+      menuChildren: (item is SubMenuItem)
+          ? item.children.map(_mapItemToButton).toList()
+          : null,
+      onSelected: (item is SubMenuItem) ? null : onTap,
+    );
+  }
+}
+
+class _MenuButton extends StatelessWidget {
+  final String label;
+  final Icon? icon;
+  final bool showDivider;
+  final bool enabled;
+  final List<Widget>? menuChildren;
+  final ValueChanged<String>? onSelected;
+
+  const _MenuButton({
+    required this.label,
+    this.icon,
+    this.showDivider = false,
+    this.enabled = true,
+    this.menuChildren,
+    this.onSelected,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final textTheme = theme.textTheme;
+
+    final textStyle = textTheme.bodyMedium?.copyWith(
+      color: enabled ? textTheme.bodySmall?.color : theme.disabledColor,
+    );
+
+    final children = menuChildren;
+    return Wrap(
+      children: [
+        Stack(
+          children: [
+            if (icon != null)
+              Positioned.fill(
+                child: Align(
+                  alignment: Alignment.centerLeft,
+                  child: Padding(
+                    padding: const EdgeInsets.only(left: 10),
+                    child: Icon(
+                      icon!.icon,
```

Comment:
How about `MenuItem` having `IconData` instead of `Icon` (widget) so here you could just build it yourself. If thats bad idea maybe simple `IconTheme` would be enough

```dart
if (icon != null)
  Positioned.fill(
    child: Align(
      alignment: Alignment.centerLeft,
      child: Padding(
        padding: const EdgeInsets.only(left: 10),
        child: IconTheme(
          data: IconThemeData(
            color: enabled
                ? textTheme.bodySmall?.color
                : theme.disabledColor,
          ),
          child: icon!,
        ),
      ),
    ),
  ),
```